cassovary-core/src/main/scala/com/twitter/cassovary/graph/Traverser.scala

```diff
@@ -396,7 +396,7 @@ trait QueueBasedTraverser extends Traverser {
 /**
  * A traverser that keeps track of first depth of visiting a given node.
  */
-trait DepthTracker extends QueueBasedTraverser {
+trait DepthTracker[NodeType <: Node] extends QueueBasedTraverser[NodeType] {
```

Comment:
I wondering if we cannot use just
```
trait DepthTracker extends QueueBasedTraverser[Node]
```
as DepthTracker itself does not use information about particular Node type.